spot_rl_experiments/spot_rl/utils/heuristic_nav.py

```diff
@@ -0,0 +1,458 @@
+                point_in_local_3d, None, spot, vision_T_hand, body_T_hand
+            )
+            if self.visualize:
+                if corner_det_status:
+                    rgb_img_vis = cv2.circle(
+                        rgb_img_vis,
+                        (best_corner_yx[-1], best_corner_yx[0]),
+                        7,
+                        (0, 0, 255),
+                        thickness=-1,
+                    )
+                    for yx in other_best_yxs:
+                        rgb_img_vis = cv2.circle(
+                            rgb_img_vis, (yx[-1], yx[0]), 1, (255, 0, 0)
+                        )
+                org = (0, y1_int - 50)
+                text = "({:.2f}, {:.2f}, {:.2f}), O:{:.2f}, Z:{:.2f}".format(
+                    global_x, global_y, np.degrees(theta), offset, z
+                )
+                rgb_img_vis = cv2.putText(
+                    rgb_img_vis,
+                    text=text,
+                    org=org,
+                    fontFace=cv2.FONT_HERSHEY_SIMPLEX,
+                    fontScale=0.9,
+                    color=(0, 0, 255),
+                    thickness=2,
+                )
+            return True, (global_x, global_y, theta), rgb_img_vis
+
+        return False, (None, None, None), rgb_img_vis
+
+
+def navigate_to_aria_goal(
+    x: float,
+    y: float,
+    theta: float,
+    spotskillmanager,
+    object_target: str = "ball",
+    image_search: ImageSearch = None,
+    save_cone_search_images: bool = True,
+    pull_back: bool = True,
+):
+    print(f"Original Nav targets {x, y, theta}")
+    if save_cone_search_images:
+        previously_saved_images = glob("imagesearch*.png")
+        for f in previously_saved_images:
+            os.remove(f)
+
+    if image_search is None:
+        image_search = ImageSearch(
+            corner_static_offset=0.5, use_yolov8=False, visualize=True
+        )
+    spotskillmanager.nav_controller.nav_env.enable_nav_goal_change()
+    (x, y) = (
+        pull_back_point_along_theta_by_offset(x, y, theta, 0.2) if pull_back else (x, y)
+    )
+    print(f"Nav targets adjusted on the theta direction ray {x, y, np.degrees(theta)}")
+    backup_steps = spotskillmanager.nav_controller.nav_env.max_episode_steps
+    spotskillmanager.nav_controller.nav_env.max_episode_steps = 50
+    spotskillmanager.nav(x, y, theta)
+    spotskillmanager.nav_controller.nav_env.max_episode_steps = backup_steps
+    spot: Spot = spotskillmanager.spot
+    spot.open_gripper()
+    gaze_arm_angles = deepcopy(spotskillmanager.pick_config.GAZE_ARM_JOINT_ANGLES)
+    spot.set_arm_joint_positions(np.deg2rad(gaze_arm_angles), 1)
+    time.sleep(1.2)
+    found, (x, y, theta), visulize_img = image_search.search(
+        object_target, *get_me_arguments_for_image_search_fn(spot, 0)
+    )
+    rate = 20
+    if not found:
+        # start semi circle search
+        for i_a, angle in enumerate(np.arange(-90, 110, 20)):
```

Comment:
could you have the variable names for -90, 110, and 20?

Reply:
that's a good idea, I have added that